include/cudaq/Optimizer/CodeGen/Pipelines.h

```diff
@@ -21,60 +21,28 @@
 
 namespace cudaq::opt {
 
-/// @brief Pipeline builder to convert Quake to QIR. `convertTo` should be
-/// specified if `QIRProfile` is true.
-/// @param QIRProfile whether or not this is lowering to a specific QIR profile
-/// @param pm Pass manager to append passes to
-/// @param convertTo String name of QIR profile (e.g., `qir-base`,
-/// `qir-adaptive`)
-template <bool QIRProfile = false>
-void addPipelineToQIR(mlir::PassManager &pm,
-                      llvm::StringRef convertTo = "none") {
-  pm.addNestedPass<mlir::func::FuncOp>(
-      cudaq::opt::createApplyControlNegations());
-  cudaq::opt::addAggressiveEarlyInlining(pm);
-  pm.addPass(mlir::createCanonicalizerPass());
-  pm.addNestedPass<mlir::func::FuncOp>(cudaq::opt::createUnwindLoweringPass());
-  pm.addPass(mlir::createCanonicalizerPass());
-  pm.addPass(cudaq::opt::createApplyOpSpecializationPass());
-  pm.addPass(cudaq::opt::createExpandMeasurementsPass());
-  pm.addNestedPass<mlir::func::FuncOp>(cudaq::opt::createClassicalMemToReg());
-  pm.addPass(mlir::createCanonicalizerPass());
-  pm.addPass(mlir::createCSEPass());
-  pm.addNestedPass<mlir::func::FuncOp>(cudaq::opt::createLowerToCFGPass());
-  pm.addNestedPass<mlir::func::FuncOp>(cudaq::opt::createQuakeAddDeallocs());
-  pm.addPass(cudaq::opt::createLoopNormalize());
-  cudaq::opt::LoopUnrollOptions luo;
-  luo.allowBreak = convertTo.equals("qir-adaptive");
-  pm.addPass(cudaq::opt::createLoopUnroll(luo));
-  pm.addPass(mlir::createCanonicalizerPass());
-  pm.addPass(mlir::createCSEPass());
-  pm.addNestedPass<mlir::func::FuncOp>(
-      cudaq::opt::createCombineQuantumAllocations());
-  pm.addPass(mlir::createCanonicalizerPass());
-  pm.addPass(mlir::createCSEPass());
-  if (convertTo.equals("qir-base"))
-    pm.addNestedPass<mlir::func::FuncOp>(
-        cudaq::opt::createDelayMeasurementsPass());
-  pm.addPass(mlir::createConvertMathToFuncs());
-  pm.addPass(cudaq::opt::createConvertToQIRPass());
-  if constexpr (QIRProfile) {
-    cudaq::opt::addQIRProfilePipeline(pm, convertTo);
-  }
-}
+/// The common pipeline.
+/// Adds the common pipeline (with or without a profile specifier) but without
+/// the final QIR profile lowering passes.
+void commonPipelineConvertToQIR(
+    mlir::PassManager &pm, const std::optional<mlir::StringRef> &convertTo);
```

Comment:
Nit: any reason to have a reference to an already reference-like object (StringRef)?
```suggestion
    mlir::PassManager &pm, const std::optional<mlir::StringRef> convertTo);
```

Reply:
Prevents making an unneeded copy of the std::optional value.